Tests/tests/test_SystemServicesTest.cpp

```diff
@@ -0,0 +1,272 @@
+        EXPECT_EQ(response,string("{\"success\":true}"));
+}
+
+TEST_F(SystemServicesTest, RebootDelay)
+{
+        EXPECT_CALL(rfcApiImplMock, setRFCParameter(::testing::_, ::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](char *pcCallerID, const char* pcParameterName, const char* pcParameterValue, DATA_TYPE eDataType) {
+                EXPECT_EQ(strcmp(pcCallerID, "thunderapi"), 0);
+                EXPECT_EQ(strcmp(pcParameterName, "Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.AutoReboot.fwDelayReboot"), 0);
+                return WDMP_SUCCESS;
+        }));
+
+    EXPECT_EQ(Core::ERROR_NONE, handlerV2.Invoke(connection, _T("setFirmwareRebootDelay"), _T("{\"delaySeconds\":10}"),response));
+    EXPECT_EQ(response,string("{\"success\":true}"));
+}
+
+TEST_F(SystemServicesTest, Firmware)
+{
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("updateFirmware"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"success\":true}"));
+	
+    EXPECT_EQ(Core::ERROR_NONE, handlerV2.Invoke(connection, _T("getLastFirmwareFailureReason"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"failReason\":\"None\",\"success\":true}"));
+
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getDownloadedFirmwareInfo"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"currentFWVersion\":\"AX013AN_5.6p4s1_VBN_sey\",\"downloadedFWVersion\":\"\",\"downloadedFWLocation\":\"\",\"isRebootDeferred\":false,\"success\":true}"));
+
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getFirmwareDownloadPercent"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"downloadPercent\":-1,\"success\":true}"));
+
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getFirmwareUpdateState"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"firmwareUpdateState\":0,\"success\":true}"));
+}
+
+TEST_F(SystemServicesTest, Timezone)
+{
+
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setTimeZoneDST"), _T("{\"timeZone\":\"America/New_York\"}"),response));
+	EXPECT_EQ(response,string("{\"success\":true}"));
+
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getTimeZoneDST"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"timeZone\":\"America\\/New_York\",\"success\":true}"));
+
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setTimeZoneDST"), _T("{\"timeZone\":\"America/Costa_Rica\"}"),response));
+    EXPECT_EQ(response,string("{\"success\":true}"));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getTimeZoneDST"), _T("{}"),response));
+    EXPECT_EQ(response,string("{\"timeZone\":\"America\\/Costa_Rica\",\"success\":true}"));
+
+	EXPECT_EQ(Core::ERROR_NONE, handlerV2.Invoke(connection, _T("getTimeZones"), _T("{}"),response));
+
+	size_t pos = response.length() - 15 ;
+    string timeZones = response.substr(pos,response.length());
+
+    EXPECT_EQ(timeZones,string("\"success\":true}"));
```

Comment:
please use `EXPECT_THAT(response, ::testing::MatchesRegex`

Reply:
This require more changes in Regex, so will be updated in next set of API's

Reply:
as of now removed this API